cdc/puller/ddl_puller_test.go

```diff
@@ -108,28 +114,144 @@ func (m *mockPuller) appendResolvedTs(ts model.Ts) {
 	})
 }
 
-func newMockDDLJobPuller(puller Puller) DDLJobPuller {
-	return &ddlJobPullerImpl{puller: puller, outputCh: make(chan *model.DDLJobEntry, defaultPullerOutputChanSize)}
+func newMockDDLJobPuller(t *testing.T, puller Puller, needSchemaSnap bool) (DDLJobPuller, *entry.SchemaTestHelper) {
+	res := &ddlJobPullerImpl{
+		puller: puller,
+		outputCh: make(
+			chan *model.DDLJobEntry,
+			defaultPullerOutputChanSize),
+		metricDiscardedDDLCounter: discardedDDLCounter.
+			WithLabelValues("ddl", "test"),
+	}
+	var helper *entry.SchemaTestHelper
+	if needSchemaSnap {
+		helper = entry.NewSchemaTestHelper(t)
+		kvStorage := helper.Storage()
+		ts := helper.GetCurrentMeta().StartTS
+		meta, err := kv.GetSnapshotMeta(kvStorage, ts)
+		require.Nil(t, err)
+		schemaSnap, err := schema.NewSingleSnapshotFromMeta(meta, ts, false)
+		require.Nil(t, err)
+		res.schemaSnapshot = schemaSnap
+		res.kvStorage = kvStorage
+	}
+	return res, helper
+}
+
+func TestHandleRenameTables(t *testing.T) {
+	startTs := uint64(10)
+	mockPuller := newMockPuller(t, startTs)
+	ddlJobPuller, helper := newMockDDLJobPuller(t, mockPuller, true)
+	defer helper.Close()
+
+	ddlJobPullerImpl := ddlJobPuller.(*ddlJobPullerImpl)
+	cfg := config.GetDefaultReplicaConfig()
+	cfg.Filter.Rules = []string{"test1.t1", "test1.t2", "test2.t4"}
+	f, err := filter.NewFilter(cfg, "")
+	require.NoError(t, err)
+	ddlJobPullerImpl.filter = f
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go ddlJobPuller.Run(ctx)
+	go func() {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ddlJobPuller.Output():
+		}
+	}()
+
+	// filter out test1.t3, but nor filter out test1.t1, test1.t2
+	{
+		job := helper.DDL2Job("create database test1")
+		mockPuller.appendDDL(job)
+		mockPuller.appendResolvedTs(job.BinlogInfo.FinishedTS + 1)
+		waitResolvedTs(t, ddlJobPuller, job.BinlogInfo.FinishedTS+1)
+
+		job = helper.DDL2Job("create table test1.t1(id int)")
+		mockPuller.appendDDL(job)
+		mockPuller.appendResolvedTs(job.BinlogInfo.FinishedTS + 1)
+		waitResolvedTs(t, ddlJobPuller, job.BinlogInfo.FinishedTS+1)
+
+		job = helper.DDL2Job("create table test1.t2(id int)")
+		mockPuller.appendDDL(job)
+		mockPuller.appendResolvedTs(job.BinlogInfo.FinishedTS + 1)
+		waitResolvedTs(t, ddlJobPuller, job.BinlogInfo.FinishedTS+1)
+
+		job = helper.DDL2Job("create table test1.t3(id int)")
+		mockPuller.appendDDL(job)
+		mockPuller.appendResolvedTs(job.BinlogInfo.FinishedTS + 1)
+		waitResolvedTs(t, ddlJobPuller, job.BinlogInfo.FinishedTS+1)
+
+		job = helper.DDL2Job("rename table test1.t1 to test1.t11, test1.t2 to test1.t22, test1.t3 to test1.t33")
```

Comment:
Add test for `rename table test1.t33 to test1.t1`.

Reply:
ok